### PR TITLE
Make tests resilient to arbitrary text properties in completions

### PR DESCRIPTION
**Pull request by joaotavora**
* test/capf-tests.el (cl-lib): Require it.
(company--remove-but-these-properties): New helper.
(company-basic-capf-highlighting)
(company-non-prefix-fancy-capf-highlighting)
(company-non-prefix-modest-capf-highlighting): Use it.

### CODE REVIEW

File: test/capf-tests.el [hunk 1]
```diff
@@ -55,6 +56,19 @@
...
     (should (member "with-current-buffer" company-candidates))))
 
+(defun company--remove-but-these-properties (string keep)
+  "Remove from STRING all text properties but the ones in KEEP."
+  (remove-list-of-text-properties
+   0 (length string)
+   (cl-set-difference
+    (cl-loop for start = 0 then (next-property-change start string)
+             while start
+             append (loop for (k _v) on (text-properties-at start string)
```
Comment (dgutov) [inline test/capf-tests.el]: cl-loop, you mean :)

**Comment by dgutov** [comment]
I was going to simply remove `completion-score`, but this should be more resilient indeed.

File: test/capf-tests.el (hunk 1 above)
Comment (joaotavora) [reply test/capf-tests.el]: oops, I mean cl-oops

File: test/capf-tests.el [hunk 2]
```diff
@@ -55,6 +56,19 @@
...
     (should (member "with-current-buffer" company-candidates))))
 
+(defun company--remove-but-these-properties (string keep)
+  "Remove from STRING all text properties but the ones in KEEP."
+  (remove-list-of-text-properties
+   0 (length string)
+   (cl-set-difference
+    (cl-loop for start = 0 then (next-property-change start string)
+             while start
+             append (cl'loop for (k _v) on (text-properties-at start string)
```
Comment (dgutov) [inline test/capf-tests.el]: Almost :)

Comment (joaotavora) [reply test/capf-tests.el]: goddamn...

**Comment by dgutov** [comment]
Thank you!

**Pull request merged**